### PR TITLE
reduce xds proxy log noise

### PR DESCRIPTION
**Pull request by ramaraochavali**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by howardjohn — CHANGES_REQUESTED** [review]
I think this is racy, HandleUpstream can be concurrent

File: pkg/istio-agent/xds_proxy.go [hunk 1]
```diff
@@ -272,10 +273,19 @@ func (p *XdsProxy) HandleUpstream(ctx context.Context, con *ProxyConnection, xds
 	upstream, err := xds.StreamAggregatedResources(ctx,
 		grpc.MaxCallRecvMsgSize(defaultClientMaxReceiveMessageSize))
 	if err != nil {
-		proxyLog.Errorf("failed to create upstream grpc client: %v", err)
+		p.connectedMutex.Lock()
```
Comment (howardjohn) [inline pkg/istio-agent/xds_proxy.go:276]: Could just use atomic.Int32? might be a bit simpler

**Review by howardjohn — APPROVED** [review]

File: pkg/istio-agent/xds_proxy.go (hunk 1 above)
Comment (ramaraochavali) [reply pkg/istio-agent/xds_proxy.go:276]: sure

**Comment by howardjohn** [comment]
I removed the "fixes" https://github.com/istio/istio/issues/29957, I think we can still make some followups once  https://github.com/envoyproxy/envoy/pull/14616 merges

**Comment by mandarjog** [comment]
Can we add a definitive message that states "connected to istiod" ?

**Comment by howardjohn** [comment]
Do you think we need an envoy -> proxy connected logs and proxy -> istiod?
Or just the latter? I am thinking probably just the latter?

They are always 1:1 by the way, assuming now errors.

**Pull request merged**